### DISCUSSION TITLE
Network policy event logging

### DISCUSSION THREAD

**Discussion by robmoore-i** [k3s-io/k3s#6109] [Q&A]
Hi 👋 I'm looking for some support with debugging network policies.

I'm running k3s on ubuntu 2020.04 (focal). I'm currently installing k3s like this: `curl -sfL https://get.k3s.io | bash -`.

I have an application which creates some NetworkPolicy objects, to control access between some containers that run in the same namespace.

I'm debugging a problem with them that's causing a container to unexpectedly be unable to communicate with another. In the course of debugging this, I would like to be able to see some logs which describe when network policies have been enforced, to block/allow a connection, and for what reason. By doing `k get netpol` I can see some information e.g. about applied pod selectors, but this is not enough in my case. I am looking for event-level logs.

If these logs are already available, somewhere, then where can I find them?
If they aren't but I can gain that ability with a special installation, then could you describe that?
If it's simply not possible, then please let me know.

Many thanks,
Rob

---
Thanks to a fab colleague of mine, the original problem I was debugging is solved. (The network policies were specifying the ports corresponding to a service, rather than the ports corresponding to a pod. NetworkPolicy acts on the pod level.)

Nevertheless, I suspect I would solved this faster if I had been able to see some logs. In this case, it would have been the lack of "connection blocked" logs that would have tipped me off to something else going on.

**Comment by brandond** [comment]
> I would like to be able to see some logs which describe when network policies have been enforced, to block/allow a connection, and for what reason.

K3s embeds the [kube-router](https://www.kube-router.io/) network policy engine, which does not offer this functionality.

Calico Enterprise is the only network policy engine that I'm aware of that will log blocks. Istio may also offer this when used with the CNI plugin but I am not very familiar with it so I can't confirm that.
